pkg/helm/locate.go

```diff
@@ -133,10 +133,14 @@ func cacheFilePath(chartURL string, cv *repo.ChartVersion, cacheDir string) (str
 	if strings.Contains(path, "..") {
 		return "", errors.Errorf("get %s: path %q points outside the cache dir", chartURL, path)
 	}
+	digest := "none"
 	if len(cv.Digest) < 16 {
-		return "", errors.Errorf("repo index entry for chart %q does not specify a digest", cv.Name)
+		// not all the helm repository implementations populate the digest field (e.g. Nexus 3)
+		log.Printf("repo index entry for chart %q does not specify a digest", cv.Name)
+	} else {
+		digest = cv.Digest[:16]
 	}
 	hostSegment := strings.ReplaceAll(u.Host, ":", "_")
-	digestSegment := fmt.Sprintf("%s-%s-%s", cv.Name, cv.Version, cv.Digest[:16])
+	digestSegment := fmt.Sprintf("%s-%s-%s", cv.Name, cv.Version, digest)
 	return filepath.Join(cacheDir, hostSegment, filepath.Dir(path), digestSegment, filepath.Base(path)), nil
```

Comment:
Maybe we need to download the tarball and compute the hash locally? Otherwise the same cache dir will be used even if the artifact changes right? (but only if the version does not also change, so maybe we do not care)

Reply:
Using a static value as fallback in case the repo index does not specify a digest per chart makes sense to me as well.